### PR TITLE
Quicksort in Scala

### PR DESCRIPTION
**Pull request by veevidify**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: archive/s/scala/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.txt
```
Comment (jrg94) [inline archive/s/scala/.gitignore]: What's this .gitignore for?

File: archive/s/scala/quick-sort.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import scala.io.StdIn.readLine
+import scala.reflect.ClassTag
+
+object QuickSortSamples {
+  def main(args: Array[String]) {
+    // example of sorting integers
```
Comment (jrg94) [inline archive/s/scala/quick-sort.scala]: These are great samples here! Unfortunately, we have special requirements for each type of sample program. You can see the description for QuickSort here: https://sample-programs.therenegadecoder.com/

We're actually in the process of writing test go to verify new samples, so this solution should follow the Quick Sort table if possible.

File: archive/s/scala/.gitignore (hunk 1 above)
Comment (veevidify) [reply archive/s/scala/.gitignore]: Shoot! I was using that to exclude txt files as input output from my git diff. I will make another change that take it out.

File: archive/s/scala/quick-sort.scala (hunk 2 above)
Comment (veevidify) [reply archive/s/scala/quick-sort.scala]: I'm a bit lost on the home page haha. Can I get a direct link to the quick sort spec?

Also for in-development tests, will they be part of CI, so contributors can to use them on our own fork's travis?

Comment (jrg94) [reply archive/s/scala/quick-sort.scala]: My bad! I thought I gave the right link: https://sample-programs.therenegadecoder.com/projects/quick-sort/

Also, yes! They're already built into Travis, but there's limited support at the moment languages other than Python and Bash.

Comment (veevidify) [reply archive/s/scala/quick-sort.scala]: yeah gotcha, quick question for some clarifications:
```
$ ./quick-sort.lang "4, 5, 3, 1, 4, 2"
1, 2, 3, 4, 4, 5
```
so input/output format is a string, which is a comma-delimited list of integers?

Comment (jrg94) [reply archive/s/scala/quick-sort.scala]: Yup! That's the idea.

**Pull request merged**